react_paragraphs.install

```diff
@@ -0,0 +1,368 @@
+}
+
+/**
+ * Get the list of entities that need to be migrated.
+ *
+ * @return array
+ *   Array of entity type, bundle and fields to migrate data.
+ *
+ * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+ * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+ */
+function _react_paragraphs_update_8205_get_entities() {
+  $entity_type_manager = \Drupal::entityTypeManager();
+
+  /** @var \Drupal\Core\Entity\EntityFieldManagerInterface $entityFieldManager */
+  $entityFieldManager = \Drupal::service('entity_field.manager');
+  $react_fields = $entityFieldManager->getFieldMapByFieldType('react_paragraphs');
+  $entities = [];
+  foreach ($react_fields as $entity_type => $fields) {
+    foreach (array_keys($fields) as $field_name) {
+      $entity_ids = $entity_type_manager->getStorage($entity_type)
+        ->getQuery()
+        ->condition($field_name, 0, '>')
+        ->execute();
+
+      foreach ($entity_ids as $entity_id) {
+        $entities["$entity_type:$entity_id:$field_name"] = "$entity_type:$entity_id:$field_name";
+      }
+    }
+  }
+  asort($entities);
+  return array_values($entities);
+}
+
+/**
+ * Modify the field storage and field configs.
+ */
+function react_paragraphs_update_8207() {
+  /** @var \Drupal\Core\Entity\EntityFieldManagerInterface $entityFieldManager */
+  $entityFieldManager = \Drupal::service('entity_field.manager');
+  $react_fields = $entityFieldManager->getFieldMapByFieldType('react_paragraphs');
+  $entity_type_manager = \Drupal::entityTypeManager();
+
+  $database = \Drupal::database();
+  $config_factory = \Drupal::configFactory();
+  /** @var \Drupal\Core\KeyValueStore\KeyValueFactoryInterface $key_value_factory */
+  $key_value_factory = \Drupal::service('keyvalue');
+  foreach ($react_fields as $entity_type => $fields) {
+    $storage = $entity_type_manager->getStorage($entity_type);
+    $key_value = $key_value_factory->get('entity.definitions.bundle_field_map')
+      ->get($entity_type);
+
+    foreach ($fields as $field_name => $bundles_info) {
+      if (isset($key_value[$field_name])) {
+        $key_value[$field_name]['type'] = 'entity_reference_revisions';
+      }
+      /** @var \Drupal\field\FieldStorageConfigInterface $field_storage */
+      $field_storage = FieldStorageConfig::load("$entity_type.$field_name");
+
+      $tables[] = $storage->getTableMapping()->getFieldTableName($field_name);
+      $tables[] = $storage->getTableMapping()
+        ->getDedicatedRevisionTableName($field_storage);
+      $table_data = [];
+
+      foreach ($tables as $table) {
+        $database->schema()->dropField($table, "{$field_name}_settings");
+
+        $query = $database->select($table, 't')->fields('t')->execute();
+        while ($row = $query->fetchAssoc()) {
+          $table_data[$table][] = $row;
+        }
+        $database->truncate($table)->execute();
```

Comment:
It would be nice to leave a warning in the docblock that this update is destructive and you highly recommend testing and having backups.

Reply:
isn't that the idea of a "Major" release?